lxc/console.go

```diff
@@ -0,0 +1,177 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"syscall"
+
+	"github.com/gorilla/websocket"
+
+	"github.com/lxc/lxd/client"
+	"github.com/lxc/lxd/lxc/config"
+	"github.com/lxc/lxd/shared/api"
+	"github.com/lxc/lxd/shared/gnuflag"
+	"github.com/lxc/lxd/shared/i18n"
+	"github.com/lxc/lxd/shared/logger"
+	"github.com/lxc/lxd/shared/termios"
+)
+
+type consoleCmd struct {
+	tty    int
+	escape string
+}
+
+func (c *consoleCmd) showByDefault() bool {
+	return true
+}
+
+func (c *consoleCmd) usage() string {
+	return i18n.G(
+		`Usage: lxc console [<remote>:]<container> [-t] [-e]
+
+Interact with the container's console devices and log.
+
+The default is to attach to the containers /dev/console. To exit the console
+type Ctrl+<escape> q. The default for <escape> is 'a'.`)
+}
+
+func (c *consoleCmd) flags() {
+	gnuflag.IntVar(&c.tty, "tty", 0, i18n.G("The console device to attach to"))
```

Comment:
As above, LXD containers only have /dev/console enabled and I don't think we want to encourage people running more ttys, so I'd recommend we keep that out for now.